src/main/java/duke/commands/Parser.java

```diff
@@ -0,0 +1,40 @@
+package duke.commands;
+
+import duke.exceptions.DukeException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Parses a command using regex and returns the corresponding command, value, flag and additionalValues
+ */
+public class Parser {
+    private static final Pattern commandParser =
+            Pattern.compile("^([a-zA-Z]+)(?:\\s([^/]+))?(?:\\s/([a-zA-Z]+))?(?:\\s(.*))?$");
+
+    public Command getCommand(String input) throws DukeException {
+        String firstWord = input.split(" ", 2)[0];
+        Command command;
+        try {
+            command = Command.valueOf(firstWord.toUpperCase());
+        } catch (IllegalArgumentException e) {
+            throw new DukeException("No Such command!");
+        }
+        return command;
+    }
+
+    public List<String> parseCommand(String command) {
+        Matcher matcher = Parser.commandParser.matcher(command);
+        matcher.find();
+        String value = matcher.group(2);
+        String flag = matcher.group(3);
+        String additionalValue = matcher.group(4);
+        List<String> result = new ArrayList<>();
```

Comment:
Should this be "results" instead since it is a list and the name of variable should be plural
```suggestion
        List<String> results = new ArrayList<>();
```

Reply:
Thank you for pointing this out!